[feat] 공개 옷장 리스트 조회 API 구현

zerone1202 opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 --> 

- Closes #21 
공개 옷장 리스트를 최신순으로 전체 조회할 수 있는 API를 구현했습니다.
## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

- 공개 여부가 true이고 삭제 여부가 false인 옷장만 전체 조회
- 오프셋 기반 페이지네이션 적용
- 정렬 기준 및 방향 설정 가능 (기본값: 최신순)
- 빈 리스트도 정상 응답 처리
- Command/Query 분리로 조회 로직을 별도 서비스로 관리
- @Transactional(readOnly = true)로 조회 성능 최적화
- record의 기본 생성자 활용으로 불필요한 Builder 제거

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)## 💬 공유사항 to 리뷰어<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

- 인증인가 구현 전, 옷장 데이터 있을 때 응답 (로그인 구현 시, 리팩토링 후 다시 테스트 예정)
<img width="1249" height="895" alt="스크린샷 2025-10-19 16 08 35" src="https://github.com/user-attachments/assets/ce077dc5-8a49-45cb-9977-011c350f1e32" />
<img width="1249" height="895" alt="스크린샷 2025-10-19 16 08 43" src="https://github.com/user-attachments/assets/2c190a0b-14eb-40df-946e-0e9b193f533b" />

- 옷장 데이터 없을 때 빈리스트 응답(인증인가 구현 전,후)
<img width="1249" height="485" alt="스크린샷 2025-10-19 16 09 29" src="https://github.com/user-attachments/assets/11a81295-2c43-4c74-9408-61e7597ff6ed" />

- URL 잘못 입력 시 응답
<img width="1249" height="485" alt="스크린샷 2025-10-19 16 09 44" src="https://github.com/user-attachments/assets/696398bc-bc37-44b6-92bb-47858e8fc46a" />


## ✅ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

zerone1202 committed:
[feat] closet: ClosetGetPublicResponse DTO 생성

- 공개 옷장 리스트 조회 응답 DTO 추가
- record 타입으로 불변 객체 구현
- 정적 팩토리 메서드 from()으로 Entity → DTO 변환
- Builder 대신 record 기본 생성자 활용

zerone1202 committed:
[feat] closet: ClosetRepository에 공개 옷장 조회 메서드 추가

- findAllByIsPublicTrueAndIsDeletedFalse 메서드 추가
- 페이지네이션
- 조건: isPublic = true AND isDeleted = false
- 공개되고 삭제되지 않은 전체 옷장만 조회

zerone1202 committed:
[feat] closet: ClosetSuccessCode에 공개 옷장 조회 성공 코드 추가

- CLOSET_GET_PUBLIC_OK 성공 코드 추가
- HttpStatus: 200 OK
- 메시지: "공개 옷장 리스트를 조회했습니다."

zerone1202 committed:
[feat] closet: 공개 옷장 리스트 조회 API 구현

- GET /api/v1/closets/public 엔드포인트 추가
- 페이지네이션 파라미터 지원
  - page (기본값: 0)
  - size (기본값: 10)
  - sort (기본값: createdAt)
  - direction (기본값: DESC)
- ApiPageResponse로 페이지 정보 포함하여 응답
- ClosetSuccessCode.CLOSET_GET_PUBLIC_OK 사용
- 200 OK 응답

zerone1202 committed:
[feat] closet: ClosetQueryService 인터페이스 생성

- 옷장 쿼리 서비스 인터페이스 정의
- getPublicClosets() 메서드 선언
  - 파라미터: page, size, sort, direction
  - 반환: Page<ClosetGetPublicResponse>
- Command/Query 분리 원칙 적용 (CQRS)

zerone1202 committed:
[feat] closet: ClosetQueryServiceImpl 구현체 생성

- ClosetQueryService 인터페이스 구현
- getPublicClosets() 메서드 구현
- 페이지네이션 로직 처리
  - Sort 객체 생성 (direction, sort)
  - PageRequest 객체 생성 (page, size, sortObj)
- Repository 호출로 공개 옷장 조회
  - isPublic = true AND isDeleted = false 조건
- Entity → DTO 변환 (map 사용)
- @Transactional(readOnly = true)로 읽기만 가능으로 설정

zerone1202 committed:
Merge branch 'dev' of https://github.com/Princess-Maker-s-OOT/OOT-Outfit-of-Today into feat/closet-21
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

AllSungho commented on src/main/java/org/example/ootoutfitoftoday/domain/closet/controller/ClosetController.java:
```diff
@@ -2,31 +2,40 @@
 @RequestMapping("/v1/closets")
 public class ClosetController {
 
     private final ClosetCommandService closetCommandService;
```
해당 Service 인터페이스는 같은 도메인의 인터페이스이므로
보다 직접적인 구현체를 사용하는 것이 더 좋다고 생각합니다!

zerone1202 replied:
한 수 배웠습니다! 성호님 최고에요👍🏻

zerone1202 commented on src/main/java/org/example/ootoutfitoftoday/domain/closet/controller/ClosetController.java:
```diff
@@ -2,31 +2,40 @@
+
+    // 공개 옷장 리스트 조회
+    @GetMapping("/public")
+    public ResponseEntity<ApiPageResponse<ClosetGetPublicResponse>> getPublicClosets(@RequestParam(defaultValue = "0") int page, @RequestParam(defaultValue = "10") int size, @RequestParam(defaultValue = "createdAt") String sort, @RequestParam(defaultValue = "DESC") String direction) {
+
+        Page<ClosetGetPublicResponse> closetGetPublicResponses = closetQueryService.getPublicClosets(page, size, sort, direction);
+
+        return ApiPageResponse.success(closetGetPublicResponses, ClosetSuccessCode.CLOSET_GET_PUBLIC_OK);
+    }
```
저희 컨벤션과 맞지 않아 수정없이 그대로 가겠습니다~

zerone1202 committed:
[feat] closet: ClosetController에서 Service 구현체 직접 주입으로 변경

- ClosetQueryService 인터페이스 → ClosetQueryServiceImpl 구현체 직접 주입
- 같은 도메인 내에서만 사용하므로 인터페이스 불필요

zerone1202 committed:
[feat] closet: 구현체 직접 주입으로 변경하여 ClosetQueryService에서 로직 제거

- ClosetQueryService 인터페이스 → ClosetQueryServiceImpl 구현체 직접 주입
- 같은 도메인 내에서만 사용하므로 인터페이스 불필요

cjn404 commented on src/main/java/org/example/ootoutfitoftoday/domain/closet/controller/ClosetController.java:
```diff
@@ -2,31 +2,40 @@
+
+    // 공개 옷장 리스트 조회
+    @GetMapping("/public")
+    public ResponseEntity<ApiPageResponse<ClosetGetPublicResponse>> getPublicClosets(@RequestParam(defaultValue = "0") int page, @RequestParam(defaultValue = "10") int size, @RequestParam(defaultValue = "createdAt") String sort, @RequestParam(defaultValue = "DESC") String direction) {
```
콘트롤러 파라미터 매핑 부분 문단 띄우기 필요해보입니다~~

cjn404 commented on src/main/java/org/example/ootoutfitoftoday/domain/closet/service/query/ClosetQueryService.java:
```diff
@@ -0,0 +1,9 @@
+
+public interface ClosetQueryService {
+
+    Page<ClosetGetPublicResponse> getPublicClosets(int page, int size, String sort, String direction);
```
파라미터 문단 띄우기 컨벤션 적용 부탁드려요~

AllSungho reviewed (approved):
완벽합니다 수고하셨습니다!
